src/sentry/remote_subscriptions/consumers/result_consumer.py

```diff
@@ -79,20 +138,95 @@ def decode_payload(self, payload: KafkaPayload | FilteredPayload) -> T | None:
             )
         return None
 
+    def create_with_partitions(
+        self,
+        commit: Commit,
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        if self.parallel:
+            return self.create_thread_parallel_worker(commit)
+        else:
+            return self.create_serial_worker(commit)
+
+    def create_serial_worker(self, commit: Commit) -> ProcessingStrategy[KafkaPayload]:
+        return RunTask(
```

Comment:
you're already using RunTask. wouldn't it be easier to conditionally swap it for RunTaskInThreads instead of maintaining your own threadpool?

Reply:
We have special logic here where we want to process a batch of results before processing the next batch. This allows us to make sure that related data is processed serially.

Reply:
This is because we need to partition the batches in such a way that uptime results for the same monitor are processed in order.

Using `RunTaskInThreads` doesn't offer any kind of way to specify ordering

Reply:
We're actually doing this in 3 different places now (crons, uptime, issue occrences). I really would love to turn this into some kind of reusable strategy

Reply:
Got it. I keep forgetting about this, feels like I asked this before 😅 

I think the part where you need to wait for a set of messages before processing the next one can be generalized independently of the inner strategy even.